pkg/kubelet/prober/worker_test.go

```diff
@@ -72,135 +73,160 @@ func TestDoProbe(t *testing.T) {
-	}
+		if !doProbe(m, w) {
+			t.Errorf("[%s] Expected to continue, but did not", probeType)
+		}
 
-	// 100 seconds later...
-	laterStatus := getRunningStatus()
-	laterStatus.ContainerStatuses[0].State.Running.StartedAt.Time =
-		time.Now().Add(-100 * time.Second)
-	m.statusManager.SetPodStatus(w.pod, laterStatus)
+		result, ok := m.resultsManager(probeType).Get(containerID)
+		if probeType == readiness {
+			if !ok {
+				t.Errorf("Expected readiness to be false during initial delay, but was not set")
+			} else if result {
+				t.Errorf("Expected readiness to be false during initial delay, but was true")
+			}
+		} else { // probeType == liveness
+			if ok {
+				t.Errorf("Expected liveness to be unset during initial delay, but was set")
+			}
+		}
 
-	// Second call should succeed (already waited).
-	if !doProbe(m, w) {
-		t.Errorf("Expected to continue, but did not")
-	}
+		// 100 seconds later...
+		laterStatus := getRunningStatus()
+		laterStatus.ContainerStatuses[0].State.Running.StartedAt.Time =
+			time.Now().Add(-100 * time.Second)
+		m.statusManager.SetPodStatus(w.pod, laterStatus)
 
-	ready, ok = m.readinessCache.Get(containerID)
-	if !ok {
-		t.Errorf("Expected readiness to be true, but was not set")
-	} else if !ready {
-		t.Errorf("Expected readiness to be true, but was false")
+		// Second call should succeed (already waited).
+		if !doProbe(m, w) {
+			t.Errorf("[%s] Expected to continue, but did not", probeType)
+		}
+
+		result, ok = m.resultsManager(probeType).Get(containerID)
+		if !ok {
+			t.Errorf("[%s] Expected result to be true, but was not set", probeType)
+		} else if !result {
+			t.Errorf("[%s] Expected result to be true, but was false", probeType)
+		}
 	}
 }
 
 func TestCleanUp(t *testing.T) {
 	m := newTestManager()
-	pod := getTestPod(api.Probe{})
-	m.statusManager.SetPodStatus(&pod, getRunningStatus())
-	m.readinessCache.Set(containerID, results.Success)
-	w := m.newWorker(&pod, pod.Spec.Containers[0])
-	m.readinessProbes[containerPath{podUID, containerName}] = w
-
-	if ready, _ := m.readinessCache.Get(containerID); !ready {
-		t.Fatal("Expected readiness to be true.")
-	}
 
-	close(w.stop)
-	if err := waitForWorkerExit(m, []containerPath{{podUID, containerName}}); err != nil {
-		t.Fatal(err)
-	}
+	for _, probeType := range [...]probeType{liveness, readiness} {
+		key := probeKey{podUID, containerName, probeType}
+		pod := getTestPod(api.Probe{})
+		m.statusManager.SetPodStatus(&pod, getRunningStatus())
+		m.resultsManager(probeType).Set(containerID, results.Success)
+		w := m.newWorker(&pod, pod.Spec.Containers[0], probeType)
+		m.workers[key] = w
 
-	if _, ok := m.readinessCache.Get(containerID); ok {
-		t.Error("Expected readiness to be cleared.")
-	}
-	if _, ok := m.readinessProbes[containerPath{podUID, containerName}]; ok {
-		t.Error("Expected worker to be cleared.")
+		if ready, _ := m.resultsManager(probeType).Get(containerID); !ready {
+			t.Fatalf("[%s] Expected result to be true.", probeType)
```

Comment:
The `.String()` method is automatically called in these places. I think it behaves kind of like java's `toString()`